从旧版本升级到 2.20，无法初始化

### 系统信息

None

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

在全新的环境下，如果先运行过 2.19（不初始化），然后再运行 2.20 之后，无法正常初始化。

![Image](https://github.com/user-attachments/assets/a8b9ad93-9a85-416c-b156-32d12a7bae72)

/kind bug
/area core

### 复现步骤

1. 使用 Docker Compose 运行 2.19
2. 停止容器，升级到 2.20
3. 访问初始化页面进行初始化

### 相关日志输出

```release-note

halo-1  | 2024-10-17T13:53:21.854+08:00  WARN 7 --- [eReconciler-t-1] io.r2dbc.h2.H2Connection                 : Option{name='readOnly', sensitive=false} + isn't supported in H2 at the transaction level. You must set it on conenction URL. See http://www.h2database.com/html/features.html#read_only
halo-1  | 2024-10-17T13:53:21.857+08:00  WARN 7 --- [eReconciler-t-1] io.r2dbc.h2.H2Connection                 : Option{name='readOnly', sensitive=false} + isn't supported in H2 at the transaction level. You must set it on conenction URL. See http://www.h2database.com/html/features.html#read_only
halo-1  | 2024-10-17T13:53:21.860+08:00  WARN 7 --- [eReconciler-t-1] io.r2dbc.h2.H2Connection                 : Option{name='readOnly', sensitive=false} + isn't supported in H2 at the transaction level. You must set it on conenction URL. See http://www.h2database.com/html/features.html#read_only
halo-1  | 2024-10-17T13:53:21.863+08:00  WARN 7 --- [eReconciler-t-1] io.r2dbc.h2.H2Connection                 : Option{name='readOnly', sensitive=false} + isn't supported in H2 at the transaction level. You must set it on conenction URL. See http://www.h2database.com/html/features.html#read_only
halo-1  | 2024-10-17T13:53:21.866+08:00  WARN 7 --- [eReconciler-t-1] io.r2dbc.h2.H2Connection                 : Option{name='readOnly', sensitive=false} + isn't supported in H2 at the transaction level. You must set it on conenction URL. See http://www.h2database.com/html/features.html#read_only
halo-1  | 2024-10-17T13:53:34.768+08:00  WARN 7 --- [ctor-http-nio-5] io.r2dbc.h2.H2Connection                 : Option{name='readOnly', sensitive=false} + isn't supported in H2 at the transaction level. You must set it on conenction URL. See http://www.h2database.com/html/features.html#read_only
halo-1  | 2024-10-17T13:53:34.919+08:00 ERROR 7 --- [oundedElastic-8] org.thymeleaf.TemplateEngine             : [THYMELEAF][boundedElastic-8] Exception processing throttled template "error/error": Exception evaluating SpringEL expression: "menuFinder.getPrimary()" (template: "modules/header" - line 5, col 3)
halo-1  |
halo-1  | org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "menuFinder.getPrimary()" (template: "modules/header" - line 5, col 3)
halo-1  | 	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at run.halo.app.theme.ReactiveSpelVariableExpressionEvaluator.evaluate(ReactiveSpelVariableExpressionEvaluator.java:36) ~[classes/:2.20.3]
halo-1  | 	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.standard.processor.StandardWithTagProcessor.doProcess(StandardWithTagProcessor.java:93) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.Model.process(Model.java:300) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.StandaloneElementTagModelProcessable.process(StandaloneElementTagModelProcessable.java:114) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ProcessorTemplateHandler.queueProcessable(ProcessorTemplateHandler.java:2106) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:1179) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.Model.process(Model.java:300) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.OpenElementTagModelProcessable.process(OpenElementTagModelProcessable.java:110) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ProcessorTemplateHandler.queueProcessable(ProcessorTemplateHandler.java:2106) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1559) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:155) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ThrottledTemplateProcessor.process(ThrottledTemplateProcessor.java:235) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.engine.ThrottledTemplateProcessor.process(ThrottledTemplateProcessor.java:200) ~[thymeleaf-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.spring6.SpringWebFluxTemplateEngine$StreamThrottledTemplateProcessor.process(SpringWebFluxTemplateEngine.java:720) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at org.thymeleaf.spring6.SpringWebFluxTemplateEngine.lambda$createChunkedStream$2(SpringWebFluxTemplateEngine.java:269) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	at reactor.core.publisher.FluxGenerate$GenerateSubscription.slowPath(FluxGenerate.java:271) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxGenerate$GenerateSubscription.request(FluxGenerate.java:213) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.request(FluxPeekFuseable.java:144) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.requestUpstream(FluxSubscribeOn.java:131) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.onSubscribe(FluxSubscribeOn.java:124) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onSubscribe(FluxPeekFuseable.java:178) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxGenerate.subscribe(FluxGenerate.java:85) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:68) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.run(FluxSubscribeOn.java:194) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[na:na]
halo-1  | 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
halo-1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
halo-1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
halo-1  | 	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
halo-1  | Caused by: run.halo.app.infra.exception.NotFoundException: 404 NOT_FOUND "No primary menu found"
halo-1  | 	at run.halo.app.theme.finders.impl.MenuFinderImpl.lambda$getPrimary$5(MenuFinderImpl.java:67) ~[classes/:2.20.3]
halo-1  | 	at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onComplete(FluxSwitchIfEmpty.java:82) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:155) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$BaseFluxToMonoOperator.completePossiblyEmpty(Operators.java:2097) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoCollectList$MonoCollectListSubscriber.onComplete(MonoCollectList.java:118) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onComplete(MonoFlatMapMany.java:261) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:152) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:152) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.drainAsync(FluxFlattenIterable.java:371) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.drain(FluxFlattenIterable.java:724) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.onComplete(FluxFlattenIterable.java:273) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onComplete(FluxPeekFuseable.java:277) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$BaseFluxToMonoOperator.completePossiblyEmpty(Operators.java:2098) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoCollectList$MonoCollectListSubscriber.onComplete(MonoCollectList.java:118) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFilter$FilterSubscriber.onComplete(FluxFilter.java:166) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onComplete(FluxMap.java:275) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$UsingWhenSubscriber.deferredComplete(FluxUsingWhen.java:397) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$CommitInner.onComplete(FluxUsingWhen.java:532) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators.complete(Operators.java:137) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoEmpty.subscribe(MonoEmpty.java:46) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$UsingWhenSubscriber.onComplete(FluxUsingWhen.java:389) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.onComplete(FluxConcatMapNoPrefetch.java:241) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onComplete(Operators.java:2231) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$UsingWhenSubscriber.deferredComplete(FluxUsingWhen.java:397) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$CommitInner.onComplete(FluxUsingWhen.java:532) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onComplete(Operators.java:2231) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:260) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onComplete(Operators.java:2231) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:210) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:210) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.pool.SimpleDequePool.maybeRecycleAndDrain(SimpleDequePool.java:540) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.pool.SimpleDequePool$QueuePoolRecyclerInner.onComplete(SimpleDequePool.java:781) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.core.publisher.Operators.complete(Operators.java:137) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoEmpty.subscribe(MonoEmpty.java:46) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.pool.SimpleDequePool$QueuePoolRecyclerMono.subscribe(SimpleDequePool.java:893) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:241) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:204) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxPeek$PeekSubscriber.onComplete(FluxPeek.java:260) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators.complete(Operators.java:137) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoEmpty.subscribe(MonoEmpty.java:46) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:265) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:241) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:204) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreElements$IgnoreElementsSubscriber.onComplete(MonoIgnoreElements.java:89) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete(FluxHandle.java:223) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onComplete(FluxSwitchIfEmpty.java:85) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoCallable$MonoCallableSubscription.request(MonoCallable.java:159) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2367) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2241) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoCallable.subscribe(MonoCallable.java:48) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:265) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreElements$IgnoreElementsSubscriber.onError(MonoIgnoreElements.java:84) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapSubscriber.onError(FluxMap.java:134) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFilter$FilterSubscriber.onError(FluxFilter.java:157) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFilter$FilterConditionalSubscriber.onError(FluxFilter.java:291) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onError(FluxMap.java:265) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoDeferContextual.subscribe(MonoDeferContextual.java:55) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$UsingWhenSubscriber.onComplete(FluxUsingWhen.java:389) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.checkTerminated(FluxFlatMap.java:850) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:612) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:592) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onComplete(FluxFlatMap.java:469) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onComplete(FluxOnAssembly.java:549) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:144) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:152) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxIterable$IterableSubscription.slowPath(FluxIterable.java:357) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxIterable$IterableSubscription.request(FluxIterable.java:294) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request(FluxMapFuseable.java:171) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapSubscriber.request(FluxMap.java:164) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.request(FluxOnAssembly.java:649) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onSubscribe(FluxFlatMap.java:373) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onSubscribe(FluxOnAssembly.java:633) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapSubscriber.onSubscribe(FluxMap.java:92) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe(FluxMapFuseable.java:96) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:201) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:83) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Flux.subscribe(Flux.java:8891) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:202) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap.subscribeOrReturn(FluxFlatMap.java:94) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Flux.subscribe(Flux.java:8876) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen$ResourceSubscriber.onNext(FluxUsingWhen.java:198) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:122) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxRetry$RetrySubscriber.onNext(FluxRetry.java:88) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondComplete(MonoFlatMap.java:245) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onNext(MonoFlatMap.java:305) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at io.r2dbc.pool.MonoDiscardOnCancel$MonoDiscardOnCancelSubscriber.onNext(MonoDiscardOnCancel.java:92) ~[r2dbc-pool-1.0.1.RELEASE.jar:1.0.1.RELEASE]
halo-1  | 	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.complete(MonoIgnoreThen.java:294) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onNext(MonoIgnoreThen.java:188) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:237) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:204) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreElements$IgnoreElementsSubscriber.onComplete(MonoIgnoreElements.java:89) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete(FluxHandle.java:223) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onComplete(FluxSwitchIfEmpty.java:85) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoCallable$MonoCallableSubscription.request(MonoCallable.java:159) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2367) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2241) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoCallable.subscribe(MonoCallable.java:48) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:265) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at io.r2dbc.pool.MonoDiscardOnCancel.subscribe(MonoDiscardOnCancel.java:50) ~[r2dbc-pool-1.0.1.RELEASE.jar:1.0.1.RELEASE]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:165) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.pool.AbstractPool$Borrower.deliver(AbstractPool.java:471) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.pool.SimpleDequePool.lambda$drainLoop$8(SimpleDequePool.java:375) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.core.scheduler.ImmediateScheduler.schedule(ImmediateScheduler.java:52) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.pool.SimpleDequePool.drainLoop(SimpleDequePool.java:375) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.pool.SimpleDequePool.pendingOffer(SimpleDequePool.java:618) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.pool.SimpleDequePool.doAcquire(SimpleDequePool.java:301) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.pool.AbstractPool$Borrower.request(AbstractPool.java:425) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.request(MonoFlatMap.java:194) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2367) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2241) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:117) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.pool.SimpleDequePool$QueueBorrowerMono.subscribe(SimpleDequePool.java:736) ~[reactor-pool-1.1.0-M5.jar:1.1.0-M5]
halo-1  | 	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxRetry$RetrySubscriber.resubscribe(FluxRetry.java:120) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoRetry.subscribeOrReturn(MonoRetry.java:50) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4560) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onError(MonoFlatMap.java:180) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxMap$MapSubscriber.onError(FluxMap.java:134) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoDeferContextual.subscribe(MonoDeferContextual.java:55) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen.subscribe(FluxUsingWhen.java:105) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Flux.subscribe(Flux.java:8891) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:202) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMapMany.subscribeOrReturn(MonoFlatMapMany.java:49) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Flux.subscribe(Flux.java:8876) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:202) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.MonoFlatMapMany.subscribeOrReturn(MonoFlatMapMany.java:49) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Flux.subscribe(Flux.java:8876) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.FluxUsingWhen.subscribe(FluxUsingWhen.java:94) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at reactor.core.publisher.Mono.blockOptional(Mono.java:1860) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 	at run.halo.app.infra.utils.ReactiveUtils.blockReactiveValue(ReactiveUtils.java:45) ~[classes/:na]
halo-1  | 	at run.halo.app.infra.utils.ReactiveUtils.blockReactiveValue(ReactiveUtils.java:28) ~[classes/:na]
halo-1  | 	at run.halo.app.theme.dialect.EvaluationContextEnhancer$ReactiveMethodExecutor.lambda$execute$1(EvaluationContextEnhancer.java:206) ~[classes/:2.20.3]
halo-1  | 	at java.base/java.util.Optional.map(Unknown Source) ~[na:na]
halo-1  | 	at run.halo.app.theme.dialect.EvaluationContextEnhancer$ReactiveMethodExecutor.execute(EvaluationContextEnhancer.java:206) ~[classes/:2.20.3]
halo-1  | 	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:152) ~[spring-expression-6.2.0-RC1.jar:6.2.0-RC1]
halo-1  | 	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:398) ~[spring-expression-6.2.0-RC1.jar:6.2.0-RC1]
halo-1  | 	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:97) ~[spring-expression-6.2.0-RC1.jar:6.2.0-RC1]
halo-1  | 	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:116) ~[spring-expression-6.2.0-RC1.jar:6.2.0-RC1]
halo-1  | 	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:338) ~[spring-expression-6.2.0-RC1.jar:6.2.0-RC1]
halo-1  | 	at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:265) ~[thymeleaf-spring6-3.1.2.RELEASE.jar:3.1.2.RELEASE]
halo-1  | 	... 43 common frames omitted
halo-1  | 	Suppressed: java.lang.Exception: #block terminated with an error
halo-1  | 		at reactor.core.publisher.BlockingOptionalMonoSubscriber.blockingGet(BlockingOptionalMonoSubscriber.java:171) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 		at reactor.core.publisher.Mono.blockOptional(Mono.java:1861) ~[reactor-core-3.7.0-M6.jar:3.7.0-M6]
halo-1  | 		... 54 common frames omitted
```

### 附加信息

_No response_

--- a/application/src/main/java/run/halo/app/theme/service/ThemeServiceImpl.java
+++ b/application/src/main/java/run/halo/app/theme/service/ThemeServiceImpl.java
@@ -47,6 +47,7 @@ import run.halo.app.extension.ReactiveExtensionClient;
 import run.halo.app.extension.Unstructured;
 import run.halo.app.infra.SystemVersionSupplier;
 import run.halo.app.infra.ThemeRootGetter;
+import run.halo.app.infra.exception.ThemeAlreadyExistsException;
 import run.halo.app.infra.exception.ThemeUpgradeException;
 import run.halo.app.infra.exception.UnsatisfiedAttributeValueException;
 import run.halo.app.infra.properties.HaloProperties;
@@ -84,6 +85,10 @@ public class ThemeServiceImpl implements ThemeService {
             ))
             .onErrorResume(IOException.class, e -> {
                 log.warn("Failed to initialize theme from {}", location, e);
+                return Mono.empty();
+            })
+            .onErrorResume(ThemeAlreadyExistsException.class, e -> {
+                log.warn("Failed to initialize theme from {}, because it already exists", location);
                 return Mono.empty();
             })
             .then();

